### PR TITLE
fix(feature/cvsb-18511): implementation to centralise aws app logging…

### PR DESCRIPTION
**Pull request by alleypa**
Implementation to centralize app logging to aws

## Ticket title
One line description
[link to ticket number](https://jira.dvsacloud.uk/browse/CVSB-18511)

## Checklist
- [x] Code has been tested manually
- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -37,13 +37,21 @@ describe('Component: Root', () => {
...
   let firebaseLogsService: FirebaseLogsService;
 
   beforeEach(async(() => {
-    syncServiceSpy = jasmine.createSpy('SyncService', () => 'startSync');
+    syncServiceSpy = jasmine.createSpyObj('SyncService', {
+      checkForUpdate: () => true
+    });
+
+    logProviderSpy = jasmine.createSpyObj('LogsProvider', {
+      dispatchLog: () => true
```
Comment (JonLazariniDD) [inline src/app/app.component.spec.ts:53]: as discussed as this is not part of this PR and scope, let's create a ticket so we can mock the store and stub the store$.dispatch method when dispatchLog is called.
This would possibly require some refactoring.

improvement ticket: https://jira.dvsacloud.uk/browse/CVSB-18639

File: src/modules/logs/logs.service.ts [hunk 2]
```diff
@@ -22,4 +25,8 @@ export class LogsProvider {
     }
     return this.httpService.sendUnauthenticatedLogs(unauthLogs);
   };
+
```
Comment (JonLazariniDD) [inline src/modules/logs/logs.service.ts:28]: I was thinking that it would be nice if we could add a unit test file for this file and dispatchLog.
I can see that we have a logs.service.mock file in place.
It is probably worth discussing it, what do you think?

File: src/pages/testing/test-creation/test-cancel/test-cancel.ts [hunk 3]
```diff
@@ -141,31 +139,34 @@ export class TestCancelPage {
...
       stack.push(
         this.testResultService.submitTestResult(testResult).pipe(
           catchError((error: any) => {
-            const log: Log = {
-              type: 'error',
-              message: `${this.oid} - ${error.status} ${
-                error.error.errors ? error.error.errors[0] : error.error
-              } for API call to ${error.url} with the body message ${JSON.stringify(testResult)}`,
+            this.logProvider.dispatchLog({
+              type: LOG_TYPES.ERROR,
```
Comment (JonLazariniDD) [inline src/pages/testing/test-creation/test-cancel/test-cancel.ts:149]: 👍

**Review by JonLazariniDD — COMMENTED** [review]
looks good, we will need to run the regression pack too

File: src/pages/testing/vehicle/vehicle-lookup/vehicle-lookup.ts [hunk 4]
```diff
@@ -117,75 +111,86 @@ export class VehicleLookupPage {
...
+              .subscribe(testHistoryResponseObserver)
+              .add(() => {
+                LOADING.dismiss();
+              });
+          }
+        },
+        (error) => {
+          this.logProvider.dispatchLog({
+            type: 'error-vehicleService.getTestResultsHistory-searchVehicle in vehicle-lookup.ts',
+            message: `${this.oid} - ${error.status} ${error.error} for API call to ${error.url}`,
```
Comment (alex-deloitte) [inline src/pages/testing/vehicle/vehicle-lookup/vehicle-lookup.ts:181]: I see it was previously error.error || error.message. I'd use JSON.stringify here also

**Review by alex-deloitte — APPROVED** [review]

File: src/app/app.component.spec.ts (hunk 1 above)
Comment (JonLazariniDD) [reply src/app/app.component.spec.ts:53]: @alleypa I had a chat with Mike (@bristol-mike-jones) earlier and it was decided that we are ok with it and would be tackled later as part of refactoring if necessary since this ticket is high priority and changes could increase the scope of this ticket.

**Review by JonLazariniDD — APPROVED** [review]
discussed with Mike earlier and agreed to proceed as the ticket is high priority and 18639 will require to be prioritised.

File: src/app/app.component.spec.ts (hunk 1 above)
Comment (JonLazariniDD) [reply src/app/app.component.spec.ts:53]: as discussed (https://github.com/dvsa/cvs-app-mobile/pull/388#discussion_r504139328) resolving.

File: src/modules/logs/logs.service.ts (hunk 2 above)
Comment (JonLazariniDD) [reply src/modules/logs/logs.service.ts:28]: as discussed (https://github.com/dvsa/cvs-app-mobile/pull/388#discussion_r504139328) resolving.
